pkg/artifactory/resource_artifactory_ldap_setting.go

```diff
@@ -66,10 +69,24 @@ func resourceArtifactoryLdapSetting() *schema.Resource {
 			Description:      "(Required) Location of the LDAP server in the following format: ldap://myldapserver/dc=sampledomain,dc=com",
 		},
 		"user_dn_pattern": {
-			Type:             schema.TypeString,
-			Required:         true,
-			ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotEmpty),
-			Description:      "(Required) A DN pattern that can be used to log users directly in to LDAP. This pattern is used to create a DN string for 'direct' user authentication where the pattern is relative to the base DN in the LDAP URL. The pattern argument {0} is replaced with the username. This only works if anonymous binding is allowed and a direct user DN can be used, which is not the default case for Active Directory (use User DN search filter instead). Example: uid={0},ou=People",
+			Type:     schema.TypeString,
+			Required: true,
+			ValidateDiagFunc: func(value interface{}, path cty.Path) diag.Diagnostics {
+				v, ok := value.(string)
+				if !ok {
+					return diag.Errorf("expected type of %q to be string", value)
+				}
+				dnPat := regexp.MustCompile(`(?i)^\s*(OU|UID|DC|CN)=(.+)`)
```

Comment:
You have the same validation function as above. Define it in Utils and reference it here 

Also, you don't need to split, you can just add the `,` to the regex with `,?`